test/Test/Database/LSMTree/Internal/IndexOrdinary.hs

```diff
@@ -0,0 +1,525 @@
+{-# LANGUAGE MagicHash #-}
+
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+{- HLINT ignore "Avoid restricted alias" -}
```

Comment:
My problem with these particular HLint rules is that using local module names like `PV` has the following issues:

* Since these names use single-letter abbreviations for words like “primitive” or “mutable”, they can be hard to understand for the uninitiated.
* These names don’t follow the standard Haskell approach of using dots to delimit components of a module name but rather tie all components (single letters in this case) directly together. From what I know, it is indeed possible to use “hierarchical” module names also as aliases introduced via `as`.

Throughout the modules I’ve been contributing, I’ve been consistently following a style that avoids these problems:

* Since virtually all list-like operations are performed with vectors in these modules, the fact that vectors are used is not mentioned in names, resulting in particular in the absence of qualification for operations using standard vectors.
* The use of special kinds of vectors is signaled by meaningful name components, like `Primitive`, resulting in readable names like `Primitive.length` or, even better, `Primitive.Vector`.

To be honest, I would have nothing against dropping the above-mentioned HLint rules.

This issue here is an instance of a more general problem I see in lots of Haskell code: People are keen to use a consistent style, which is good, but this often results in them using a suboptimal style, because this style happens to be customary and possibly encouraged or even enforced by certain tools.

Reply:
I understand your points. When working in a team, I personally prefer consistent style over optimal style (whatever that means), but in this project we are generally not very strict on that.

I'm personally in favour of adhering to the hlint rules, but I can't speak for the others.